### PR TITLE
Update URL dynamically when switching worksheets

### LINKED ISSUE #1457: cl work worksheet without reloading entire page (need to change URL)

### PR DESCRIPTION
**Pull request by kashizui**
With support for using the back/forward buttons. Things move much faster now!

Fixes #1457 

@percyliang @pbhatnagar3 please review

### CODE REVIEW

**Comment by percyliang** [comment]
Code looks good to me.
There's a bug where if you switch to another worksheet and try to edit the title, it shows the old title.
We should test this carefully to make sure we don't introduce weird bugs.

**Comment by kashizui** [comment]
That is a vexingly difficult bug to figure out..

**Comment by kashizui** [comment]
I think this bug is related to the one where it seems like we're editing the wrong metadata field in the side bar.

**Comment by kashizui** [comment]
Fixed the bug..

@percyliang please review

File: codalab/apps/web/static/js/worksheet/editable_field.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/** @jsx React.DOM */
+
+var EditableField = React.createClass({
```
Comment (percyliang) [inline codalab/apps/web/static/js/worksheet/editable_field.js]: Thanks for doing this!

**Comment by percyliang** [comment]
LGTM

**Pull request merged**